src/editor/nodes/MediaFrameNode.js

```diff
@@ -0,0 +1,141 @@
+import {
+  Object3D,
+  BoxBufferGeometry,
+  PlaneBufferGeometry,
+  MeshBasicMaterial,
+  ShaderMaterial,
+  Mesh,
+  Vector3,
+  DoubleSide
+} from "three";
+import EditorNodeMixin from "./EditorNodeMixin";
+
+export const MediaType = {
+  ALL: "all",
+  ALL_2D: "all-2d",
+  MODEL: "model",
+  IMAGE: "image",
+  VIDEO: "video",
+  PDF: "pdf"
+};
+
+export default class MediaFrameNode extends EditorNodeMixin(Object3D) {
+  static legacyComponentName = "media-frame";
+
+  static nodeName = "Media Frame";
+
+  static _geometry = new BoxBufferGeometry();
+
+  constructor(editor) {
+    super(editor);
+
+    this.mediaTypes = MediaType.ALL_2D;
+
+    const box = new Mesh(
+      MediaFrameNode._geometry,
+      new ShaderMaterial({
+        uniforms: {
+          opacity: { value: 1 }
+        },
+        vertexShader: `
+            varying vec2 vUv;
+            void main()
+            {
+              gl_Position = projectionMatrix * modelViewMatrix * vec4( position, 1.0 );
+              vUv = uv;
+            }
+          `,
+        fragmentShader: `
+            // adapted from https://www.shadertoy.com/view/Mlt3z8
+            float bayerDither2x2( vec2 v ) {
+              return mod( 3.0 * v.y + 2.0 * v.x, 4.0 );
+            }
+            float bayerDither4x4( vec2 v ) {
+              vec2 P1 = mod( v, 2.0 );
+              vec2 P2 = mod( floor( 0.5  * v ), 2.0 );
+              return 4.0 * bayerDither2x2( P1 ) + bayerDither2x2( P2 );
+            }
+
+            varying vec2 vUv;
+            uniform float opacity;
+            void main() {
+              float alpha = max(step(0.45, abs(vUv.x - 0.5)), step(0.45, abs(vUv.y - 0.5))) - 0.5;
+              if( ( bayerDither4x4( floor( mod( gl_FragCoord.xy, 4.0 ) ) ) ) / 16.0 >= alpha * opacity ) discard;
+              gl_FragColor = vec4(1.0, 1.0, 1.0, 1.0);
+            }
+          `,
+        side: DoubleSide
+      })
+    );
+
+    const previewMaterial = new MeshBasicMaterial();
+    previewMaterial.side = DoubleSide;
+    previewMaterial.transparent = true;
+    previewMaterial.opacity = 0.5;
+
+    const previewMesh = new Mesh(new PlaneBufferGeometry(1, 1, 1, 1), previewMaterial);
+    box.add(previewMesh);
+
+    box.layers.set(1);
+
+    this.helper = box;
+    this.add(box);
+
+    this.onDeselect();
+  }
+
+  onSelect() {
+    this.helper.material.uniforms.opacity.value = 1.0;
+  }
+
+  onDeselect() {
+    this.helper.material.uniforms.opacity.value = 0.5;
+  }
+
+  copy(source, recursive = true) {
+    if (recursive) {
+      this.remove(this.helper);
+    }
+
+    this.mediaTypes = source.mediaTypes;
+
+    super.copy(source, recursive);
+
+    if (recursive) {
```

Comment:
Cargo culted this from the rest of the nodes, but don't really get whats going on here.

Reply:
The `clone()` method calls the constructor which creates your helper object. The `copy()` method also clones the child elements if the recursive parameter is passed. It is passed when cloning the scene or duplicating an object in the editor. So we need to remove the helper object before copying and then re-associate the cloned helper object after the copy method is called. This looks correct :+1: